scipy/stats/_stats_py.py

```diff
@@ -3213,25 +3214,28 @@ def gstd(a, axis=0, ddof=1):
       fill_value=999999)
 
     """
-    a = np.asanyarray(a)
-    log = ma.log if isinstance(a, ma.MaskedArray) else np.log
+    if isinstance(a, ma.MaskedArray):
```

Comment:
I think I see. What do you think about deprecating the current MaskedArray behavior then?

Reply:
Ralf has

> Once we're more confident that everything is in good shape, see if we should start emitting warnings for input types that may see a change in behavior 

under "desirable for later" on the linked tracker. We definitely want to deprecate at some point! But if making this experimental mode the default is going to be a SciPy 2.0 thing (as I think it might be shaping up to be, we still have a long way to go), then maybe there is no rush.

I suppose, no harm in deprecating here in isolation though!